### ISSUE TITLE
Infinite loop caused by Layout/MultilineMethodDefinitionBraceLayout and Layout/HeredocArgumentClosingParenthesis interaction

### ISSUE THREAD

**Issue by laurmurclar** [rubocop-hq/rubocop#8046]
I'm occasionally running into an infinite loop error when I have both `Layout/MultilineMethodDefinitionBraceLayout` and `Layout/HeredocArgumentClosingParenthesis` enabled. By "occasionally" I mean that it only happens on some files in our codebase, but it happens consistently on those files.

--------

## Original code
``` ruby
foo(
  "Some string",
  <<~TEXT,
    Lots of
    Lovely
    Text
  TEXT
  some_arg: { foo: "bar" })
```
## Expected behavior
Code transformed to:
```ruby
foo(
  "Some string",
  <<~TEXT,
    Lots of
    Lovely
    Text
  TEXT
  some_arg: { foo: "bar" }
)
```

## Actual behavior
An infinite loop error:
```
bundle exec 'rubocop --safe-auto-correct --disable-uncorrectable example.rb'
Inspecting 1 file
C

Offenses:

example.rb:8:27: C: [Corrected] Layout/HeredocArgumentClosingParenthesis: Put the closing parenthesis for a method call with a HEREDOC parameter on the same line as the HEREDOC opening.
  some_arg: { foo: "bar" })
                          ^
example.rb:8:27: C: [Corrected] Layout/MultilineMethodCallBraceLayout: Closing method call brace must be on the line after the last argument when opening brace is on a separate line from the first argument.
  some_arg: { foo: "bar" })
                          ^

0 files inspected, 2 offenses detected, 2 offenses corrected
Infinite loop detected in /Users/lauramurphy-clarkin/src/intercom/example.rb.
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:278:in `block in iterate_until_no_changes'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:274:in `loop'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:274:in `iterate_until_no_changes'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:245:in `do_inspection_loop'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:124:in `block in file_offenses'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:148:in `file_offense_cache'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:122:in `file_offenses'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:113:in `process_file'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:92:in `block in each_inspected_file'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `each'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `reduce'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:91:in `each_inspected_file'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:78:in `inspect_files'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/runner.rb:39:in `run'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/cli/command/execute_runner.rb:21:in `execute_runner'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/cli/command/execute_runner.rb:13:in `run'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/cli/command.rb:10:in `run'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/cli/environment.rb:17:in `run'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/cli.rb:65:in `run_command'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/cli.rb:72:in `execute_runners'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/lib/rubocop/cli.rb:41:in `run'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/exe/rubocop:13:in `block in <top (required)>'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/2.6.0/benchmark.rb:308:in `realtime'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rubocop-0.83.0/exe/rubocop:12:in `<top (required)>'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/bin/rubocop:23:in `load'
/Users/lauramurphy-clarkin/.rbenv/versions/2.6.5/bin/rubocop:23:in `<main>'
```

## Steps to reproduce the problem

Run `rubocop --safe-auto-correct --disable-uncorrectable example.rb` with the _Original code_ snippet in _example.rb_. Reproducible with the following cop configuration:
```yaml
AllCops:
  TargetRailsVersion: 6.0
  TargetRubyVersion: 2.6
  DisplayCopNames: true
  DisabledByDefault: true

Layout/HeredocArgumentClosingParenthesis:
  Enabled: true

Layout/MultilineMethodCallBraceLayout:
  Enabled: true
```

## RuboCop version

```
$ bundle exec rubocop -V
0.83.0 (using Parser 2.7.1.2, running on ruby 2.6.5 x86_64-darwin19)
```

**Comment by denys281** [comment]
MultilineMethodDefinitionBraceLayout autocorrects it to

```
foo(
  "Some string",
  <<~TEXT,
    Lots of
    Lovely
    Text
  TEXT
  some_arg: { foo: "bar" }
)
```

then HeredocArgumentClosingParenthesis also trying to autocorrect and result is 

```
foo(
  "Some string",
  <<~TEXT,
    Lots of
    Lovely
    Text
  TEXT
  some_arg: { foo: "bar" })
```

@koic what behavior HeredocArgumentClosingParenthesis should have in current example? Should it give offenses? or should it skip ?  Or should we add MultilineMethodDefinitionBraceLayout to autocorrect_incompatible_with method?
